module/multicast.py

```diff
@@ -0,0 +1,133 @@
+                German BOS Information Script
+                     by Bastian Schroll
+
+@file:        multicast.py
+@date:        02.05.2021
+@author:      Thierry Fredrich
+@description: Implements multicast alarms
+
+structure of a multicast alarm 
+# 1 ) network delimiter without text 
+# 2 ) alarm ric without text 
+# 3 ) text ric with message 
+# 4 ) network delimiter ric message <BS>
+"""
+import logging
+from module.moduleBase import ModuleBase
+
+
+logging.debug("- %s loaded", __name__)
+
+
+class BoswatchModule(ModuleBase):
+    """!Description of the Module"""
+    # initializing with empty list
+    
+    textRics = []
+    delimiterRic = 42
+    delimiterSubric = 42
+    ignoreTime = 42
+    ### to be cleared
+    receivedAlarmRic = 42
+    receivedAlarmSubric = 42
+    bufferListFormingAMulticastAlarm = []
+    initialDelimiterReceived = False
+    alarmReceived = False
+    textRicReceived = False
+    textRicMessage = ''
+    def initStorage(self):
+        self.receivedAlarmRic = 42
+        self.receivedAlarmSubric = 42
+        self.bufferListFormingAMulticastAlarm = []
+        self.initialDelimiterReceived = False
+        self.alarmReceived = False
+        self.textRicReceived = False
+        self.textRicMessage = ''
+    
+    def __init__(self, config):
+        """!Do not change anything here!"""
+        super().__init__(__name__, config)  # you can access the config class on 'self.config'
+        logging.debug("starting multicast module")
+        logging.debug("multicastAlarm_delimiterRic is: %i" % self.config.get("multicastAlarm_delimiterRic"))
+        logging.debug("multicastAlarm_delimiterSubric is: %i" % self.config.get("multicastAlarm_delimiterSubric"))
+        logging.debug("multicastAlarm_ignore_time is: %i" % self.config.get("multicastAlarm_ignore_time"))
+        logging.debug("multicastAlarm_textRics is: %s" % self.config.get("multicastAlarm_textRics"))
+        self.delimiterRic = int(self.config.get("multicastAlarm_delimiterRic"))
+        self.delimiterSubric = int(self.config.get("multicastAlarm_delimiterSubric"))
+        self.ignoreTime = int(self.config.get("multicastAlarm_ignore_time"))
+        for aTextRic in self.config.get("multicastAlarm_textRics").split(','):
+            self.textRics.append(int(aTextRic))
+        self.initStorage()
+
+    def onLoad(self):
+        """!Called by import of the plugin
+        Remove if not implemented"""
+        pass
+
+    def doWork(self, bwPacket):
+        """!start an run of the module.
+
+        @param bwPacket: A BOSWatch packet instance"""
+        thisRic = int(bwPacket.get("ric"))
+        thisSubric = int(bwPacket.get("subric"))-1
+        thisMessage = bwPacket.get("message")
+        if bwPacket.get("mode") == "pocsag":
+            pass
+        else:
+            logging.error("multicast module only works with pocsag")
+            raise NameError('multicast module only works with pocsag')

```

Comment:
Please respect the defined return values for the modules.
See "Rückgabewert bei Modulen".
https://docs.boswatch.de/develop/ModulPlugin.html#arbeiten-mit-dem-bwpacket

Reply:
A module should return either of these: bwPacket, False, None

According to the manual, "False" simply stops the Router. In the above situation, everything should stop because the user demands something impossible? Is raising an error a problem for the rest of the logic?

Reply:
I'm currently not sure where this exception will be caught, but I think it will kill the entire server application.
The idea behind this router concept is that you can also have multiple types of messages in a single pipeline.

So it is possible that I want to send ZVEI and POCSAG via a single pipeline (e.g. send by Telegram), but only the POCSAG files should be modified by this module.
For me this is not a error as nothing will break if you return `False`